test/e2e/storage/volume_provisioning.go

```diff
@@ -163,6 +163,77 @@ func testDynamicProvisioning(t storageClassTest, client clientset.Interface, cla
 	return pv
 }
 
+// testGlusterDynamicProvisioning tests the same workflow of testDynamicProvisioning except it wont attach
+// pv to a pod volume and try write/read. This is due to a limitation in heketi which require raw devices to
+// provision a real volume. But this test make sure heketi responded properly for the volumecreate requests
+// from the glusterfs dynamic provisioner.
+func testGlusterDynamicProvisioning(t storageClassTest, client clientset.Interface, claim *v1.PersistentVolumeClaim, class *storage.StorageClass) {
+	var err error
+	/*if class != nil {
+		By("creating a StorageClass " + class.Name)
+		class, err = client.StorageV1().StorageClasses().Create(class)
+		Expect(err).NotTo(HaveOccurred())
+		defer func() {
+			framework.Logf("deleting storage class %s", class.Name)
+			framework.ExpectNoError(client.StorageV1().StorageClasses().Delete(class.Name, nil))
+		}()
+	}
+
+	By("creating a claim")
+	claim, err = client.CoreV1().PersistentVolumeClaims(claim.Namespace).Create(claim)
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		framework.Logf("deleting claim %q/%q", claim.Namespace, claim.Name)
+		// typically this claim has already been deleted
+		err = framework.DeletePersistentVolumeClaim(client, claim.Name, claim.Namespace)
+		if err != nil {
+			framework.Failf("Error deleting claim %q. Error: %v", claim.Name, err)
+		}
+	}()
+	*/
+	err = framework.WaitForPersistentVolumeClaimPhase(v1.ClaimBound, client, claim.Namespace, claim.Name, framework.Poll, framework.ClaimProvisionTimeout)
+	Expect(err).NotTo(HaveOccurred())
+
+	By("checking the claim")
+	// Get new copy of the claim
```

Comment:
All the checks below look like copied from somewhere else. Can you please make a new function and call it instead of copying the code around?

Reply:
@jsafrane the code part is from `testDynamicProvisioning` , once the workflow is complete I will functionalize and call it from `testGlusterDynamicProvisioning`.